src/wallet/rpc/transactions.cpp

```diff
@@ -642,7 +649,7 @@ RPCHelpMan listsinceblock()
         const CWalletTx& tx = pairWtx.second;
 
         if (depth == -1 || abs(wallet.GetTxDepthInMainChain(tx)) < depth) {
-            ListTransactions(wallet, tx, 0, true, transactions, filter, nullptr /* filter_label */, /*include_change=*/include_change);
+            ListTransactions(wallet, tx, 0, true, transactions, filter, /*filter_label=*/filter_label.has_value() ? &filter_label.value() : nullptr, /*include_change=*/include_change);
```

Comment:
Maybe rebase the refactor commit as the first commit so you don't have to do dereference the std::optional value.

Reply:
Great